src/Snap/Internal/Util/FileUploads.hs

```diff
@@ -528,6 +638,69 @@ setUploadTimeout s u = u { uploadTimeout = s }
 
 
 ------------------------------------------------------------------------------
+
+data EmptyFileNamePolicy = EmptyFileNameStore Int64
```

Comment:
Just wondering why this isn't just this under the hood:

```haskell
-- better names of course ;)
data EmptyFileNamePolicy = EFN { skip :: !Bool, maxSize :: !Int64 }
```

Seems like some of the logic is simpler if you do this?